py-polars/polars/testing.py

```diff
@@ -208,13 +208,15 @@ def _assert_series_inner(
         can_be_subtracted = False
 
     check_exact = check_exact or not can_be_subtracted or left.dtype == Boolean
-
     if check_dtype:
         if left.dtype != right.dtype:
             raise_assert_detail(obj, "Dtype mismatch", left.dtype, right.dtype)
 
     # create mask of which (if any) values are unequal
     unequal = left != right
+    if unequal.any() and left.dtype in (Float32, Float64):
```

Comment:
Can do, but in asserts you're typically checking that two objects are the same ("is the produced value the same as the expected value"), rather than strictly checking that they _compare_ equal - though nobody wants to write `assert_frames_are_the_same`, so equality gets conflated with sameness ;)

Either way a flag is a good idea to allow for explicit control; how about something like `nans_are_equal` that defaults to True? Then the default case when testing is to validate "sameness", but you can opt-out for any tests that explicitly need it not to do that.

Reply:
I think it should be `False`. I understand that you often want to check for `sameness`, but I want the defaults to be correct by default.

Reply:
Done; exposed on both asserts as `nans_compare_equal` - cheers.